src/support_code_library_builder/types.ts

```diff
@@ -22,21 +22,17 @@ export interface ITestStepHookParameter {
   testStepId: string
 }
 
-export type TestCaseHookFunctionWithoutParameter = () => any | Promise<any>
-export type TestCaseHookFunctionWithParameter = (
-  arg: ITestCaseHookParameter
+export type TestCaseHookFunction = (
```

Comment:
Have you tried this?

```typescript
export type TestCaseHookFunction<World = any> = (
  this: World,
  arg?: ITestCaseHookParameter
) => any | Promise<any>
```

Reply:
That works but doesn't improve things, I think. The type of `this` would still be `any` by default.

IMO `this` should default to our builtin `World` which includes `parameters` being readonly, the `attach` signature etc. If I do that with your suggestion then the signature with a custom world becomes like:

```typescript
Before(async function() {
    this.doThing()
} as TestCaseHookFunction<CustomWorld>)
```

which is sort of a breaking change for users who are doing it the pseudo-argument way.

Reply:
Followed this path and got something that covers all the bases now - thanks!